libnestutil/block_vector.h

```diff
@@ -606,6 +640,13 @@ operator-( const const_iterator& other ) const
   return ( block_index_ - other.block_index_ ) * max_block_size + ( this_element_index - other_element_index );
 }
 
+template < typename value_type_, typename ref_, typename ptr_ >
+inline typename bv_iterator< value_type_, ref_, ptr_ >::reference bv_iterator< value_type_, ref_, ptr_ >::operator[](
```

Comment:
I am a bit confused: Stripping the template arguments, this looks to me like `bv_iterator::operator[]`. But do we really want to apply array indexing to the *iterator*? I'd expected it used on the *container*. But I may  be overlooking something here.

Reply:
For iterators, the `operator[]` is the offset dereference operator, which gets the dereferenced element shifted `n` places relative to the iterator position. It is required for random access iterators.